src/main/java/org/cryptomator/common/LazyProcessedProperties.java

```diff
@@ -0,0 +1,73 @@
+package org.cryptomator.common;
+
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Pattern;
+
+public class LazyProcessedProperties extends PropertiesDecorator {
```

Comment:
Last but not least: I don't like the "Lazy" part in the name. Just because the previous solution eagerly replaced all properties, this class isn't doing anything "lazy" in particular. The class itself behaves "normal".

In other words: Better rename to e.g. `SubstitutingProperties` (or if we make `Environment` our only source, maybe even `EnvSubstProperties`, a little tribute to the awesome Linux `envsubst` command).

Reply:
Renamed to `SubstitutingProperties` in https://github.com/cryptomator/cryptomator/pull/2957/commits/79eea4814ec361745fc8dfe3ddfc307e705b38f9